kura/org.eclipse.kura.rest.network.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/network/configuration/NetworkConfigurationRestService.java

```diff
@@ -273,4 +277,32 @@ private boolean isNetworkConfigurationPid(String pid) {
         }
         return result;
     }
+
+    private void updateNetInterfaces(Map<String, Object> configurationProperties, String pid) throws KuraException {
+        if (pid.equals(NETWORK_CONFIGURATION_SERVICE_PID)) {
+            Set<String> interfaceNames = new HashSet<>();
+            configurationProperties.keySet().forEach(key -> {
+                Optional<String> interfaceName = parseInterfaceName(key);
+                interfaceName.ifPresent(interfaceNames::add);
+            });
+            Map<String, Object> properties = this.configurationService.getComponentConfiguration(pid)
+                    .getConfigurationProperties();
+            String netInterfaces = (String) properties.get("net.interfaces");
+            List<String> netInterfacesList = Arrays.asList(netInterfaces.split(","));
+            for (String name : interfaceNames) {
+                if (!netInterfacesList.contains(name)) {
+                    netInterfaces += "," + name;
+                }
+            }
+            configurationProperties.put("net.interfaces", netInterfaces);
+        }
+    }
+
+    private Optional<String> parseInterfaceName(String key) {
+        Optional<String> interfaceName = Optional.empty();
+        if (key.startsWith("net.interface.") && key.contains(".config.")) {
+            interfaceName = Optional.of(StringUtils.substringBetween(key, "net.interface.", ".config."));
+        }
```

Comment:
I think you can avoid the `if` statement by using the `Optional.ofNullable`

```suggestion
        interfaceName = Optional.ofNullable(StringUtils.substringBetween(key, "net.interface.", ".config."));
```